sys/net/if_bridge.c

```diff
@@ -3556,8 +3557,8 @@ bridge_pfil(struct mbuf **mp, struct ifnet *bifp, struct ifnet *ifp, int dir)
 		}
 
 		/* Recalculate the ip checksum. */
-		ip = mtod(*mp, struct ip *);
-		hlen = ip->ip_hl << 2;
+		struct ip *ip = mtod(*mp, struct ip *);
+		int hlen = ip->ip_hl << 2;
 		if (hlen < sizeof(struct ip))
```

Comment:
These should be at the start of the block. Or at least declared there (make the case ETHERTYPE_IP: start a new block).